### ISSUE TITLE
python->Excel error vs user message 

### ISSUE THREAD

**Issue by sdementen** [xlwings/xlwings#242]
When an exception occurs in python code, the error message appears as a popup box in Excel (when calling python from Excel).
In my xlwings workflow, I need in fact two type of message popping to the user:
- unhandled exceptions ==> this should popup in excel like the current message (type vbCritical). Moreover, in this case, I generate also a report with a full stack trace (and the name/values of all variables in each stack) that contains enough debugging information for me to debug (on top of it, it sends it to me by email so I get this debug message from all users of the excel tool).
- handled exceptions but that should display a friendly "error message" to the user ==> I currently raise a specific exception "ExcelUserMessage". In excel, if i see such error, the popup display the message but a bit differently (type vbInformation).

It could be useful to structure a bit this workflow in xlwings as I think it could benefit users in general.
In summary:
- create specific xlwings exception : UserDataValidationError, UserGenericMessage, ... that can be raised in python and that would be displayed specifically in Excel throuhg a msgbox
- for unhandled exceptions, allow the generation of a full stack trace that would be available either in the clipboard (if ctrl-C) or in a file (that could be opened easily form excel) or managed otherwise (via email or other adhoc procedure). This would be configurable in xlwings, one could provide a "handle_error" callback that could be used to customise the error reporting mgt (like send an email).

Just some ideas but ready to provide some code (like the full stack trace with variables names/values).

**Comment by fzumstein** [comment]
Sure code is always helpful!

**Comment by fzumstein** [comment]
see https://github.com/xlwings/xlwings/pull/1605
